ownership-chain/runtime/laos/src/configs/block_rewards_source.rs

```diff
@@ -1,5 +1,106 @@
-use crate::{Runtime, RuntimeEvent};
+use crate::{Balances, Runtime};
+use frame_support::weights::Weight;
+use pallet_block_rewards_source::{BalanceOf, WeightInfo};
+use pallet_parachain_staking::PayoutCollatorReward;
+use sp_runtime::{traits::Zero, DispatchError};
+use sp_std::marker::PhantomData;
 
 impl pallet_block_rewards_source::Config for Runtime {
-	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = pallet_block_rewards_source::weights::SubstrateWeight<Runtime>;
+	type Currency = Balances;
+}
+pub struct BlockRewardsSourceAdapter<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime: pallet_parachain_staking::Config + pallet_block_rewards_source::Config>
+	PayoutCollatorReward<Runtime, BalanceOf<Runtime>> for BlockRewardsSourceAdapter<Runtime>
+{
+	fn payout_collator_reward(
+		_round_index: pallet_parachain_staking::RoundIndex,
+		collator_id: Runtime::AccountId,
+		amount: pallet_block_rewards_source::BalanceOf<Runtime>,
+	) -> Weight {
+		match pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(
+			collator_id,
+			amount.into(),
+		) {
+			Ok(amount) if amount.is_zero() => Weight::zero(),
+			Ok(_) => <Runtime as pallet_block_rewards_source::Config>::WeightInfo::send_rewards(),
+			Err(_) => Weight::zero(),
+		}
+	}
+
+	fn deposit_into_existing(
+		delegator_id: &Runtime::AccountId,
+		amount: pallet_block_rewards_source::BalanceOf<Runtime>,
+	) -> Result<pallet_block_rewards_source::BalanceOf<Runtime>, DispatchError> {
+		pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(delegator_id.clone(), amount)
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::{tests::ExtBuilder, AccountId};
+
+	#[test]
```

Comment:
there is no test case where `weight` is non-zero?

Reply:
here: https://github.com/freeverseio/laos/blob/65eca57b9c75cebb22c912d41b3f6548240f64d8/ownership-chain/runtime/laos/src/configs/block_rewards_source.rs#L67-L68
look at `assert_ne`